handlers/upload.go

```diff
@@ -103,16 +103,18 @@ func (d *CatalystAPIHandlersCollection) handleUploadVOD(w http.ResponseWriter, r
 	var requestID = config.RandomTrailer(8)
 	log.AddContext(requestID, "source", uploadVODRequest.Url)
 
-	// find source segment URL
+	// find output storage URLs for source and target segments
+	var useTargetForSourceOutput bool
 	var tURL string
 	for _, o := range uploadVODRequest.OutputLocations {
+		tURL = o.URL
```

Comment:
I've been struggling to parse this logic a bit in my head, so it could be good to split it out into a method on the VOD request like `GetSourceOutputURL()` that we can unit test. In particular, I'm trying to understand if the ordering of OutputLocations now affects which one gets used for the target / what happens in the case where no "source segments" target is set

Reply:
Updated + added unit tests for both `getSourceOutputURL()` and `getTargetURL()`.